### PR TITLE
fix(Redesign Servizi): [IARS-37] Fix disabled webhook flag when reactivating service communication

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR fixes an issues that wouldn't reactivate notification channel if user reactivates the service communication

https://user-images.githubusercontent.com/3959405/124942626-174bb900-e00c-11eb-995e-705139b67309.mp4

### CODE REVIEW

**Review by pietro909 — APPROVED** [review]
Observed the bug on my device and tested this fix on iOS emulator: works fine.

File: ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts [hunk 1]
```diff
@@ -1,13 +1,53 @@
...
 import { mapKinds } from "./handleGetServicePreferenceSaga";
 
+/**
+ * Generates the payload for the updating preferences request, if a users disables the inbox flag than the other flags
+ * are disabled.
+ * If a user activate a disabled inbox flag than webhook is enabled too.
+ * @param currentServicePreferenceState
+ * @param action
+ */
+const calulateUpdatingPreference = (
```
Comment (Undermaken) [inline ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts:24]:
```suggestion
const calculateUpdatingPreference = (
```

File: ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts [hunk 2]
```diff
@@ -1,13 +1,53 @@
...
+  servicePreferenceSelector,
+  ServicePreferenceState
+} from "../../../store/reducers/entities/services/servicePreference";
+import { isServicePreferenceResponseSuccess } from "../../../types/services/ServicePreferenceResponse";
 import { mapKinds } from "./handleGetServicePreferenceSaga";
 
+/**
+ * Generates the payload for the updating preferences request, if a users disables the inbox flag than the other flags
+ * are disabled.
+ * If a user activate a disabled inbox flag than webhook is enabled too.
```
Comment (Undermaken) [inline ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts:20]:
```suggestion
 * If a user activates a disabled inbox flag than webhook is enabled too.
```

**Review by Undermaken — APPROVED** [review]

File: ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts [hunk 3]
```diff
@@ -19,18 +59,16 @@ export function* handleUpsertServicePreference(
...
-      is_webhook_enabled: action.payload.inbox ? action.payload.push : false,
-      is_email_enabled: action.payload.inbox ? action.payload.email : false,
-      settings_version: action.payload
-        .settings_version as ServicePreference["settings_version"]
-    };
+  const currentPreferences: ReturnType<typeof servicePreferenceSelector> = yield select(
+    servicePreferenceSelector
+  );
 
+  const updatingPreference = calulateUpdatingPreference(
```
Comment (Undermaken) [inline ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts:66]:
```suggestion
  const updatingPreference = calculateUpdatingPreference(
```

**Review by Undermaken — APPROVED** [review]

**Pull request merged**